astroid/rebuilder.py

```diff
@@ -220,7 +289,11 @@ def visit_module(
             package=package,
             parent=None,
         )
-        newnode.postinit([self.visit(child, newnode) for child in node.body])
+        newnode.postinit(
+            [self.visit(child, newnode) for child in node.body],
+            doc_node=self.visit(doc_ast_node, newnode),
+        )
+        self._fix_doc_node_position(newnode)
```

Comment:
Shouldn't the `_fix_doc_node_position` be called in the `postinit`? Do we ever want to call `postinit` and not call `_fix_doc_node_position`?

Reply:
`_fix_doc_node_position` depends on `self._data` from the `TreeRebuilder`.

Reply:
Ah forgot about that. Hm yeah, passing `_data` to `postinit` also feels weird. Not really a big fan of having methods that are required to be run sequentially instead of at once, but perhaps here it is unavoidable without too convoluted solutions...

Reply:
> When I added `self._data` initially, I also though if it would make sense to store it in the nodes itself. E.g. as attribute for `Module`. In the end I decided against it. These are are parsing steps which should just be done by the `TreeRebuilder`.

Yeah, this makes more sense.

> It might not be 100% perfect, but tbh I don't really mind it in this case. It's only for the tree visit functions anyway and similar to `_get_doc` which is also called sequentially. Any alternative would be worse.

It also helps that this is likely not going to change much in the future and will be unnecessary after 3.8+.

I'll do a final review but I think this is (almost) good to go!